src/util.cpp

```diff
@@ -41,24 +41,51 @@
 #include "data.h"
 #include "settings.h"
 
-static QString collapseTemplate(const QString& str, int level)
+QString collapseTemplate(const QString& str, int level)
 {
+
     if (str.indexOf(QLatin1Char('<')) == -1) {
         return str;
     }
 
+    auto isSpaceOrAngelBracket = [](QChar c) {
+        if (c == QLatin1Char(' ')) {
+            return true;
+        }
+        if (c == QLatin1Char('<')) {
+            return true;
+        }
+        if (c == QLatin1Char('>')) {
+            return true;
+        }
+
+        return false;
+    };
+
     QString output;
     output.reserve(str.size());
+    const auto operatorKeyword = QStringLiteral("operator");
+    const int size = str.size();
     int depth = 0;
-    for (auto c : str) {
+    for (int i = 0; i < size; i++) {
+        const auto c = str[i];
         if (c == QLatin1Char('<')) {
             depth++;
             if (depth == level) {
                 output.append(QLatin1String("<..."));
             }
         } else if (c == QLatin1Char('>')) {
             depth--;
+        } else if (c == QLatin1Char('o') && str.midRef(i, operatorKeyword.size()) == operatorKeyword) {
+            i += operatorKeyword.size();
+            output.append(operatorKeyword);
+            for (; i < size && isSpaceOrAngelBracket(str[i]); i++) {
+                output.append(str[i]);
+            }
+            output.append(str[i]);
```

Comment:
why is this line here, this seems to be eating one char too many? are you trying to compensate for the `i++` in the inner and outer `for` loops? maybe just do a `--i` after the last iteration of the inner loop? or use different loop vars?

Reply:
I used a different loop var and added a comment explaining whats going on